DisCatSharp.ApplicationCommands/Workers/RegistrationWorker.cs

```diff
@@ -0,0 +1,382 @@
+                || ApplicationCommandsExtension._globalDiscordCommands == null
+            )
+                return null;
+
+            var discord = ApplicationCommandsExtension._globalDiscordCommands;
+
+            List<ulong> InvalidCommandIds = new();
+
+            if (updateList == null)
+            {
+                foreach (var cmd in discord)
+                {
+                    InvalidCommandIds.Add(cmd.Id);
+                }
+            }
+            else
+            {
+                foreach (var cmd in discord)
+                {
+                    if (!updateList.Any(ul => ul.Name == cmd.Name))
+                        InvalidCommandIds.Add(cmd.Id);
+                }
+            }
+
+            return InvalidCommandIds;
+        }
+
+        /// <summary>
+        /// Builds a list of global commands to be created on discords backend.
+        /// </summary>
+        /// <param name="updateList">The command list.</param>
+        /// <returns>A list of commands.</returns>
+        private static List<DiscordApplicationCommand> BuildGlobalCreateList(List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._globalDiscordCommands == null || !ApplicationCommandsExtension._globalDiscordCommands.Any() || updateList == null)
+                return updateList;
+
+            var discord = ApplicationCommandsExtension._globalDiscordCommands;
+
+            List<DiscordApplicationCommand> NewCommands = new();
+
+            foreach (var cmd in updateList)
+            {
+                if (discord.Any(d => d.Name == cmd.Name))
+                {
+                    NewCommands.Add(cmd);
+                }
+            }
+
+            return NewCommands;
+        }
+
+        /// <summary>
+        /// Builds a list of global commands to be overwritten on discords backend.
+        /// </summary>
+        /// <param name="updateList">The command list.</param>
+        /// <returns>A dictionary of command ids and commands.</returns>
+        private static Dictionary<ulong, DiscordApplicationCommand> BuildGlobalOverwriteList(List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._globalDiscordCommands == null || !ApplicationCommandsExtension._globalDiscordCommands.Any()
+                || updateList == null || ApplicationCommandsExtension._globalDiscordCommands == null
+            )
+                return null;
+
+            var discord = ApplicationCommandsExtension._globalDiscordCommands;
+
+            Dictionary<ulong, DiscordApplicationCommand> UpdateCommands = new();
+            foreach (var cmd in updateList)
+            {
+                if (discord.Any(d => d.Name == cmd.Name))
+                {
+                    UpdateCommands.Add(discord.Where(d => d.Name == cmd.Name).First().Id, cmd);
```

Comment:
Same comment as above, we're finding the item twice.

Reply:
still doing an any / where  --- finding it twice

Reply:
true. will change